template/styles/docfx.js

```diff
@@ -61,10 +64,29 @@ $(function () {
         } else {
           $(anchor).addClass('external-link');
         }
-    })
+      })
+  }
+
+  function addGtmButtons() {
+    $(".sample-container").each(function (i, block) {
+      const paragraph = $('<p>').attr('style','margin: 0;padding-top: 0.5rem').text("Like this sample? Get access to our complete Angular toolkit and start building your own apps in minutes. ");
+      const link=$('<a>');
+      
+      link.attr('data-xd-ga-action','Download');
+      link.attr('data-xd-ga-label','Ignite UI for Angular');
+      link.attr({
+        target:"_blank",
+        href:"https://www.infragistics.com/products/ignite-ui-angular/download",
+        class: "no-external-icon mchNoDecorate trackCTA"
+      });
+      link.text("Download it for free.").appendTo(paragraph);
+      if(i===0){
+        $(block).next().is('p') ? "" : $(block).after(paragraph)

```

Comment:
Here the check must be more specific...the parapgraph won't be added if there is another different paragraph present right after the "sample-container" element. You can try checking whether the nextSibling paragraph has an anchor child with class "trackCTA".